### PR TITLE
Sorts the outputs and also prints out alternative driver info when no driver is picked.

### LINKED ISSUE #10009: Rejected drivers not shown in priority order
`Unable to pick a default driver. Here is what was considered, in preference order:`

Unfortunately that is a lie, only the Available drivers are sorted - not the rejects...

### PR DESCRIPTION
**Pull request by lxzhang000**
fixes #10009

Looks like "Unhealthy" prioriy level is now only used for driver selection, and I think it is not necessary since we can get the unhealthy info to filter the driver out from `State.Healthy`.

Another problem associated with this one is that the current alternatives is not printed out when failed.

Before:
1) All rejects are in "Unhealthy" priority therefore can't not be displayed by preference.
2) No info of alternative drivers that are not picked.

After:
1) All rejects keeps their orginal priority.
2) The alternative drivers lower than "Discouraged" priority will also be printed out (take "ssh" as an example).
```
* Unable to pick a default driver. Here is what was considered, in preference order:
  ...
  - ssh: Rejected due to low priority
  ...

```

### CODE REVIEW

**Comment by medyagh** [comment]
@lxzhang000 please check unit test

run `make test` locally

**Review by medyagh — CHANGES_REQUESTED** [review]
I can not verify if this PR is brekaing a logic that @tstromberg intended or not (classifiying an unhealthy driver as a priority of its own)

I refer judgment to @tstromberg

File: pkg/minikube/driver/driver.go [hunk 1]
```diff
@@ -279,7 +279,11 @@ func Suggest(options []registry.DriverState) (registry.DriverState, []registry.D
 				continue
 			}
 
-			ds.Rejection = fmt.Sprintf("%s is preferred", pick.Name)
+			if pick.Name == "" {
+				ds.Rejection = "Rejected due to low priority"
```
Comment (tstromberg) [inline pkg/minikube/driver/driver.go:283]: Let's make this more specific and actionable. What do you think about this message:

`Not selected: "experimental" drivers must be specified using --driver`

**Review by tstromberg — CHANGES_REQUESTED** [review]
Overall, things seem right to me. Tested using:

`touch /tmp/podman && chmod 755 /tmp/podman && env PATH=/tmp ./out/minikube start`

Output:

```
😄  minikube v1.17.1 on Debian rodete
👎  Unable to pick a default driver. Here is what was considered, in preference order:
    ▪ docker: Not installed: exec: "docker": executable file not found in $PATH
    ▪ kvm2: Not installed: exec: "virsh": executable file not found in $PATH
    ▪ vmware: Not installed: exec: "docker-machine-driver-vmware": executable file not found in $PATH
    ▪ virtualbox: Not installed: unable to find VBoxManage in $PATH
    ▪ ssh: Rejected due to low priority
    ▪ none: Not installed: exec: "iptables": executable file not found in $PATH
    ▪ podman: Not healthy: exec: "sudo": executable file not found in $PATH

❌  Exiting due to DRV_NOT_DETECTED: No possible driver was detected. Try specifying --driver, or see https://minikube.sigs.k8s.io/docs/start/
```

The only request is to improve the "Rejected due to low priority" text.

File: pkg/minikube/driver/driver.go (hunk 1 above)
Comment (afbjorklund) [reply pkg/minikube/driver/driver.go:283]: Needs some other word than "experimental" though, the `none` and `ssh` are more like opt-in.

`Not selected: "%s" driver must be specified explicitly using --driver` or something to that effect ?

**Comment by afbjorklund** [comment]
For podman it is true that it is still experimental, but hope to change that for next release. It should be on par with Docker.

See #10237

File: pkg/minikube/driver/driver.go (hunk 1 above)
Comment (lxzhang000) [reply pkg/minikube/driver/driver.go]: Updated. Thanks for your advice.

Used "Not selected: %s must be specified explicitly using --driver" in accordance with the the previous string
"%s is preferred"

**Comment by lxzhang000** [comment]
Hi, @medyagh I have updated the failed unit tests. FYI, I could pass the uni test locally even before the modification and the reason is that if two drivers have the same priority, their order is not determined.

I suppose the root cause is the traversal order of go map is not determinded:
https://github.com/kubernetes/minikube/blob/47da1ffe3f4b45a0ffdedea1bbea2912278fda26/pkg/minikube/registry/registry.go#L158

Looks like for display it doesn't care the order if drivers have the same priority, I use different priorites for unit test.

**Pull request closed without merging**